anchor/src/processor.rs

```diff
@@ -36,7 +48,38 @@ fn process_deposit(
     amount: StLamports,
 ) -> ProgramResult {
     let accounts = DepositAccountsInfo::try_from_slice(accounts_raw)?;
+    if amount == StLamports(0) {
+        msg!("Amount must be greater than zero");
+        return Err(ProgramError::InvalidArgument);
+    }
+
     let anchor = deserialize_anchor(program_id, accounts.anchor)?;
+    if &anchor.reserve_account != accounts.to_reserve_account.key {
```

Comment:
So the reserve account is owned by the reserve authority? We could make it an associated token account. Then we don't need to store the reserve account address in the instance, and that eliminates one opportunity for mismatches. (Of course, we still need to verify that the account passed to the instruction call is the associated token account.)